src/main/java/org/skife/jdbi/v2/ClasspathStatementLocator.java

```diff
@@ -118,31 +115,42 @@ public String locate(String name, StatementContext ctx)
                 found.put(cache_key, name == cache_key ? new String(name) : name);
                 return name;
             }
-
-            final StringBuffer buffer = new StringBuffer();
-            reader = new BufferedReader(new InputStreamReader(in_stream, Charset.forName("UTF-8")));
-            String line;
+            StringBuilder sb = new StringBuilder();
             try {
-                while ((line = reader.readLine()) != null) {
-                    if (isComment(line)) {
-                        // comment
-                        continue;
+                SqlScriptLexer lexer = new SqlScriptLexer(new ANTLRInputStream(in_stream));
+                boolean endOfFile = false;
+                while (!endOfFile) {
+                    Token t = lexer.nextToken();
+                    switch (t.getType()) {
+                        case SqlScriptLexer.EOF:
+                            endOfFile = true;
+                            break;
+                        case SqlScriptLexer.COMMENT:
+                        case SqlScriptLexer.MULTI_LINE_COMMENT:
+                            break;
+                        case SqlScriptLexer.NEWLINES:
+                            if (sb.length() > 0) {
+                                sb.append(" ");
+                            }
+                            break;
+                        default:
+                            sb.append(t.getText());
```

Comment:
Is this code reachable today?  It might be preferable to cover all possibilities as `case` statements and then have the `default` throw an IllegalStateException or something.